### PR TITLE
Script to run mysql queries and upload files to s3

### PR DESCRIPTION
**Pull request by shaidar**
Run the following:
- [x] Users query
- [x] Student module query
- [x] Enrollment query
- [ ] Roles query
- [x] Upload files to S3 bucket
- [x] Notify slack

### CODE REVIEW

File: mitx_residential_etl.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    import requests
+    from logbook import Logger, RotatingFileHandler
+    from sqlalchemy import create_engine
+    from sqlalchemy.sql import text
+except ImportError as err:
+    print("Failed to import module: ", err)
+    sys.exit("Make sure to install logbook, requests and sqlalchemy")
+
+datetime = datetime.now()
+date_suffix = datetime.strftime('%m%d%Y')
```
Comment (blarghmatey) [inline mitx_residential_etl.py]: I generally find this format hard to parse. In general my preferred date format is `%Y-%m-%d`.

File: mitx_residential_etl.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+except IOError:
+    sys.exit("[-] Failed to read settings file")
+
+# Configure logbook logging
+logger = RotatingFileHandler(settings['Logs']['logfile'],
+                             max_size=int(settings['Logs']['max_size']),
+                             backup_count=int(settings['Logs']['backup_count']),
+                             level=int(settings['Logs']['level']))
+logger.push_application()
+logger = Logger(__name__)
```
Comment (blarghmatey) [inline mitx_residential_etl.py]: Is there a reason that you're overwriting the `logger` variable?

File: mitx_residential_etl.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    except SyntaxError as err:
+        logger.exception("Failed to run sync command. Check if awscli is installed")
+        notify_slack_channel(f"Failed to run sync command: `{err}`")
+        sys.exit("[-] Failed to run sync command. Check if awscli is installed")
+    except subprocess.SubprocessError as err:
+        logger.exception("Failed to sync local files to s3 bucket")
+        notify_slack_channel(f"Sync failed: `{err}`")
+        sys.exit("[-] Failed to sync daily_folder to s3 bucket")
+    logger.info("S3 sync successfully ran: {}", cmd_output)
+    notify_slack_channel(f"Sync succeeded: `str({cmd_output}`)")
```
Comment (blarghmatey) [inline mitx_residential_etl.py]: One thing to note with f strings is that Ubuntu installs 3.5 from the system repositories which doesn't have that feature. We can use it, but it will require installing Python 3.6 on the target host.

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+awscli
+logbook
+mysql-connector-python
+requests
+sqlalchemy
```
Comment (blarghmatey) [inline requirements.txt]: Since this is a greenfield project we should probably use pipenv for managing dependencies. It supports generating a requirements.txt file as well, which we will need for using with Salt.

File: mitx_residential_etl.py (hunk 3 above)
Comment (shaidar) [reply mitx_residential_etl.py]: Yeah I saw that it needed 3.6 but think it might be easier to use 3.5 as after testing it took some finagling to get 3.6 with pip3 to work and use that instead of 3.5.x. Will change in new commit.

File: requirements.txt (hunk 4 above)
Comment (shaidar) [reply requirements.txt]: I will replace the requirements.txt file with pipenvs Pipfile in the repo

File: mitx_residential_etl.py (hunk 1 above)
Comment (shaidar) [reply mitx_residential_etl.py]: Changed

File: mitx_residential_etl.py (hunk 2 above)
Comment (shaidar) [reply mitx_residential_etl.py]: No specific reason apart from cookie-cutter kinda code. Removed.

**Comment by blarghmatey** [comment]
:
